instructor/compiler.py

```diff
@@ -0,0 +1,469 @@
+from instructor.models import *
+import json
+
+
+def preview_as_json(assignment_id):
+    ret = preview(assignment_id)
+    return json.dumps(ret)
+
+
+def preview(assignment_id):
+    ret = dict()
+    ret['app_title'] = 'StarCellBio Preview'
+    ret['app_description'] = 'StarCellBio Instructor Preview'
+    ret['assignments'] = {'list': [compile(assignment_id)]}
+    return ret
+
+
+def compile(assignment_id):
+    a = Assignment.objects.get(id=assignment_id)
+    ret = {
+        'id': 'a_{}'.format(a.id),
+        'name': a.name,
+        'course': a.course.code,
+        'course_name': a.course.name,
+        'description': a.name,
+        'notebook': {},
+        'experiments': {},
+        'template': {
+            'experiment_setup': 'Experiment Setup',
+            'ui': {
+                'experimental_design': {
+                    'techniques': compile_techniques(a),
+                    'gel_types': gel_types(a)
+                },
+                'experiment_setup': {
+                    'table': [
+                        {'kind': "cell_line",
+                         'title': "Strain",
+                         'editable': 'false'
+                        },
+                    ], 'actions': [
+                    {'kind': 'add_protocol', 'name': 'Add Protocol'}
+                ]
+                }
+            },
+            'cell_lines': compile_cell_lines(a.strains.all()),
+            'time_unit': {
+                'kind': 'minutes'
+            }
+        }
+    }
+    treatments = [{'kind': "drug", 'title': "Treatments", 'editable': False}, ]
```

Comment:
this is all nuts, you really should use a template for this instead of all this madness in python